presto-prometheus/src/main/java/com/facebook/presto/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -0,0 +1,271 @@
+                    long packedValue = (long) domain.getValues().getRanges().getSpan().getLow().getValue();
+                    Instant instant = ofEpochMilli(unpackMillisUtc(packedValue));
+                    timeInfoBuilder.setPredicateLowerTimeBound(Optional.of(instant));
+                }
+                if (!domain.getValues().getRanges().getSpan().includes(Marker.upperUnbounded(TIMESTAMP_WITH_TIME_ZONE))) {
+                    long packedValue = (long) domain.getValues().getRanges().getSpan().getHigh().getValue();
+                    Instant instant = ofEpochMilli(unpackMillisUtc(packedValue));
+                    timeInfoBuilder.setPredicateUpperTimeBound(Optional.of(instant));
+                }
+            });
+            return timeInfoBuilder.build();
+        });
+    }
+
+    static String decimalSecondString(long millis)
+    {
+        return new BigDecimal(Long.toString(millis)).divide(new BigDecimal(1000L)).toPlainString();
+    }
+
+    // URIBuilder handles URI encode
+    private static URI buildQuery(URI baseURI, String time, String metricName, Duration queryChunkSizeDuration)
+            throws URISyntaxException
+    {
+        return HttpUriBuilder
+                .uriBuilderFrom(baseURI)
+                .replacePath("api/v1/query")
+                .addParameter("query", metricName + "[" + queryChunkSizeDuration.roundTo(queryChunkSizeDuration.getUnit()) +
+                        Duration.timeUnitToString(queryChunkSizeDuration.getUnit()) + "]")
+                .addParameter("time", time)
+                .build();
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transactionHandle,
+            ConnectorSession session,
+            ConnectorTableLayoutHandle layout,
+            SplitSchedulingContext splitSchedulingContext)
+    {
+        PrometheusTableLayoutHandle layoutHandle = (PrometheusTableLayoutHandle) layout;
+        PrometheusTableHandle tableHandle = layoutHandle.getTableHandle();
+        PrometheusTable table = prometheusClient.getTable(tableHandle.getSchemaName(), tableHandle.getTableName());
+
+        // this can happen if table is removed during a query
+        if (table == null) {
+            throw new TableNotFoundException(tableHandle.toSchemaTableName());
+        }
+        List<ConnectorSplit> splits = generateTimesForSplits(prometheusClock.now(), maxQueryRangeDuration, queryChunkSizeDuration, tableHandle)
+                .stream()
+                .map(time -> {
+                    try {
+                        return new PrometheusSplit(buildQuery(
+                                prometheusURI,
+                                time,
+                                table.getName(),
+                                queryChunkSizeDuration));
+                    }
+                    catch (URISyntaxException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)